crates/fluvio-controlplane-metadata/src/mirror/status.rs

```diff
@@ -20,11 +21,42 @@ impl MirrorStatus {
         last_seen: u64,
     ) -> Self {
         Self {
-            pairing,
+            pairing_sc: pairing,
+            pairing_spu: MirrorPairStatus::Waiting,
             connection_status,
             connection_stat: ConnectionStat { last_seen },
         }
     }
+
+    pub fn new_by_spu(pairing_spu: MirrorPairStatus, last_seen: u64) -> Self {
+        Self {
+            pairing_spu,
+            connection_stat: ConnectionStat { last_seen },
+            ..Default::default()
+        }
+    }
+
+    pub fn merge_from_sc(&mut self, other: Self) {
+        self.pairing_sc = other.pairing_sc;
+        self.connection_status = other.connection_status;
+        self.connection_stat = other.connection_stat;
+    }
+
+    pub fn merge_from_spu(&mut self, other: Self) {
+        self.pairing_spu = other.pairing_spu;
+        self.connection_stat = other.connection_stat;
+    }
+
+    pub fn pair_errors(&self) -> String {
+        match (self.pairing_sc.clone(), self.pairing_spu.clone()) {
```

Comment:
this is bit awkward.  clone should be done after match

Reply:
Changing `&self` to just `self` does not even require clones, fixed!